data/src/main/java/org/cryptomator/data/cloud/crypto/MasterkeyCryptoCloudProvider.kt

```diff
@@ -286,15 +290,15 @@ class MasterkeyCryptoCloudProvider(
 		}
 	}
 
-	class UnlockTokenImpl(private val vault: Vault, val keyFileData: ByteArray) : UnlockToken {
+	class UnlockTokenImpl(private val vault: Vault, private val secureRandom: SecureRandom, val keyFileData: ByteArray) : UnlockToken {
 
 		override fun getVault(): Vault {
 			return vault
 		}
 
 		@Throws(IOException::class)
 		fun getKeyFile(password: CharSequence?): Masterkey {
-			return MasterkeyFileAccess(CryptoConstants.PEPPER, SecureRandom()).load(ByteArrayInputStream(keyFileData), password)
+			return MasterkeyFileAccess(CryptoConstants.PEPPER, secureRandom).load(ByteArrayInputStream(keyFileData), password)
```

Comment:
What's the reason for this change?

Reply:
There is no need to instantiate a new `SecureRandom` for each call to `MasterkeyFileAccess.load()`